central/graphql/resolvers/platform_cve_core.go

```diff
@@ -0,0 +1,249 @@
+
+	return ret, nil
+}
+
+// PlatformCVE returns graphQL resolver for specified platform cve id.
+func (resolver *Resolver) PlatformCVE(ctx context.Context, args struct {
+	CveID              *string
+	SubfieldScopeQuery *string
+}) (*platformCVECoreResolver, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.Root, "PlatformCVE")
+
+	if !features.VulnMgmtNodePlatformCVEs.Enabled() {
+		return nil, errors.Errorf("Feature %s is disabled", features.VulnMgmtWorkloadCVEs.Name())
+	}
+	if err := readClusters(ctx); err != nil {
+		return nil, err
+	}
+	if args.CveID == nil {
+		return nil, errors.New("cveID variable must be set")
+	}
+
+	query := search.NewQueryBuilder().AddExactMatches(search.CVEID, *args.CveID).ProtoQuery()
+	if args.SubfieldScopeQuery != nil {
+		rQuery := RawQuery{
+			Query: args.SubfieldScopeQuery,
+		}
+		filterQuery, err := rQuery.AsV1QueryOrEmpty()
+		if err != nil {
+			return nil, err
+		}
+		query = search.ConjunctionQuery(query, filterQuery)
+	}
+
+	cves, err := resolver.PlatformCVEView.Get(ctx, query)
+	if len(cves) == 0 {
+		return nil, nil
+	}
+	if len(cves) > 1 {
+		utils.Should(errors.Errorf("Retrieved multiple rows when only one row is expected for CVEID=%s query", *args.CveID))
+		return nil, err
+	}
+	ret, err := resolver.wrapPlatformCVECoreWithContext(ctx, cves[0], err)
+	if err != nil {
+		return nil, err
+	}
+	ret.subFieldQuery = query
+
+	return ret, nil
+}
+
+// CVE returns the platform cve name
+func (resolver *platformCVECoreResolver) CVE(_ context.Context) string {
+	return resolver.data.GetCVE()
+}
+
+// CVEType returns the type of the given platform cve
+func (resolver *platformCVECoreResolver) CVEType(_ context.Context) string {
+	return resolver.data.GetCVEType().String()
+}
+
+func (resolver *platformCVECoreResolver) CVSS(_ context.Context) float64 {
+	return float64(resolver.data.GetCVSS())
+}
+
+// ClusterCountByPlatformType returns the number of clusters of each type affected by the given platform cve
+func (resolver *platformCVECoreResolver) ClusterCountByPlatformType(ctx context.Context) (*clusterCountByPlatformTypeResolver, error) {
+	return resolver.root.wrapClusterCountByPlatformTypeWithContext(ctx, resolver.data.GetClusterCountByPlatformType(), nil)
+}
+
+// ClusterCount returns the number of clusters affected by the given platform cve
+func (resolver *platformCVECoreResolver) ClusterCount(_ context.Context) int32 {
+	return int32(resolver.data.GetClusterCount())
+}
+
+// Clusters returns a paginated list of clusters containing given platform cve
+func (resolver *platformCVECoreResolver) Clusters(ctx context.Context, args struct{ Pagination *inputtypes.Pagination }) ([]*clusterResolver, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.PlatformCVECore, "Clusters")
+
+	if err := readClusters(ctx); err != nil {
+		return nil, err
+	}
+
+	query := search.NewQueryBuilder().AddExactMatches(search.CVE, resolver.data.GetCVEID()).ProtoQuery()
```

Comment:
En, shall it be (search.CVEID, resolver.data.GetCVEID()) or (search.CVE, resolver.data.GetCVE())?

Reply:
That is right it should be search.CVEID